transform-plugins/src/test/java/co/cask/hydrator/plugin/XMLParserTest.java

```diff
@@ -0,0 +1,338 @@
+      "year:/book/year", "title:string,author:int,year:int", "Exit on error");
+    try {
+      Transform<StructuredRecord, StructuredRecord> transform = new XMLParser(config);
+      transform.initialize(new MockTransformContext());
+
+      StructuredRecord inputRecord = StructuredRecord.builder(INPUT)
+        .set("offset", 1)
+        .set("body", "<book category=\"COOKING\"><title lang=\"en\">Everyday Italian</title>" +
+          "<author>Giada De Laurentiis</author><year>2005</year><price>30.00</price></book>").build();
+
+      MockEmitter<StructuredRecord> emitter = new MockEmitter<>();
+      transform.transform(inputRecord, emitter);
+    } catch (IllegalStateException e) {
+      LOG.error("Test passes. Exiting on exception.", e);
+    }
+  }
+
+  @Test
+  public void testXpathArray() throws Exception {
+    String inputTable = "input-xpath-array";
+    ETLStage source = new ETLStage("source", MockSource.getPlugin(inputTable));
+    Map<String, String> sourceProperties = new ImmutableMap.Builder<String, String>()
+      .put("input", "body")
+      .put("encoding", "UTF-8")
+      .put("xpathMappings", "category://book/@category,title://book/title")
+      .put("fieldTypeMapping", "category:string,title:string")
+      .put("processOnError", "Exit on error")
+      .build();
+
+    ETLStage transform = new ETLStage("transform",
+                                      new ETLPlugin("XMLParser", Transform.PLUGIN_TYPE, sourceProperties, null));
+    String sinkTable = "output-xpath-array";
+
+    ETLStage sink = new ETLStage("sink", MockSink.getPlugin(sinkTable));
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(transform)
+      .addStage(sink)
+      .addConnection(source.getName(), transform.getName())
+      .addConnection(transform.getName(), sink.getName())
+      .build();
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    ApplicationId appId = NamespaceId.DEFAULT.app("XMLReaderTest");
+    ApplicationManager appManager = deployApplication(appId.toId(), appRequest);
+
+    DataSetManager<Table> inputManager = getDataset(inputTable);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(INPUT)
+        .set("offset", 1)
+        .set("body", "<bookstore><book category=\"cooking\"><title lang=\"en\">Everyday Italian</title>" +
+          "<author>Giada De Laurentiis</author><year>2005</year><price>30.00</price></book>" +
+          "<book category=\"children\"><title lang=\"en\">Harry Potter</title><author>J K. Rowling</author>" +
+          "<year>2005</year><price>29.99</price></book></bookstore>").build()
+    );
+    MockSource.writeInput(inputManager, input);
+    try {
+      MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+      mrManager.start();
+      mrManager.waitForFinish(5, TimeUnit.MINUTES);
```

Comment:
put an `Assert.fail(...);` call after the `waitForfinish`.
Currently, when the mapreduce completes (without throwing exception), the assertion on line 335 will not get executed, because no IllegalStateException is caught.